src/pudl/metadata/codes.py

```diff
@@ -4,6 +4,65 @@
 import numpy as np
 import pandas as pd
 
+ENTITY_TYPES_EIA: Dict[str, Any] = {
+    "df": pd.DataFrame(
+        columns=[
+            'code',
+            'label',
+            'description',
+        ],
+        data=[
+            ('A', 'municipal_marketing_authority', 'Municipal Marketing Authority. Voted into existence by the residents of a municipality and given authority for creation by the state government. They are nonprofit organizations'),
+            ('B', 'behind_the_meter', 'Behind the Meter. Entities that install, own, and/or operate a system (usually photovoltaic), and sell, under a long term power purchase agreement (PPA) or lease, all the production from the system to the homeowner or business with which there is a net metering agreement. Third Party Owners (TPOs) of PV solar installations use this ownership code.'),
+            ('C', 'cooperative', 'Cooperative. Member-owned organizations.'),
+            ('COM', 'commercial', 'Commercial facility.'),
+            ('D', 'nonutility_dsm_administrator',
+             'Non-utility DSM Administrator. Only involved with Demand-Side Management activities.'),
+            ('F', 'federal', 'Federal. Government agencies with the authority to deliver energy to end-use customers.'),
+            ('G', 'community_choice_aggregator', 'Community Choice Aggregator.'),
+            ('I', 'investor_owned',
+             'Investor-owned Utilities. Entities that are privately owned and provide a public service.'),
+            ('IND', 'industrial', 'Industrial facility.'),
+            ('M', 'municipal', 'Municipal: Entities that are organized under authority of state statute to provide a public service to residents of that area.'),
+            ('O', 'other', 'Other entity type.'),
+            ('P', 'political_subdivision', 'Political Subdivision. (also called "public utility district"): Independent of city or county government and voted into existence by a majority of the residents of any given area for the specific purpose of providing utility service to the voters. State laws provide for the formation of such districts.'),
+            ('PO', 'power_marketer', 'Power marketer.'),
+            ('PR', 'private', 'Private entity.'),
+            ('Q', 'independent_power_producer',
+             'Independent Power Producer or Qualifying Facility. Entities that own power plants and sell their power into the wholesale market.'),
+            ('R', 'retail_power_marketer',
+             'Retail Power Marketer or Energy Service Provider: Entities that market power to customers in restructured markets.'),
+            ('S', 'state', 'State entities that own or operate facilities or provide a public service.'),
+            ('T', 'transmission', 'Transmission: Entities that operate or own high voltage transmission wires that provide bulk power services.'),
+            ('U', 'unknown', 'Unknown entity type.'),
```

Comment:
I guess unknowns could later be made into `pd.NA` but I think since it's actually a `U` code being mapped it's okay to leave as `unknown`.

Reply:
Because the `entity_type` field ended up being much more of a dumpster 🔥 than I expected (see #1392) I ended up not using this coding table at all -- it's just a freeform string field for the moment. But I agree when we get to that issue Unknown could totally end up being `pd.NA`